extensions/sips/src/utils.ts

```diff
@@ -0,0 +1,26 @@
+import { runAppleScript } from "run-applescript";
+
+export const getSelectedImages = async () => {
+  const selectedImages = await runAppleScript(
+    `tell application "Finder"
+      set theSelection to selection
+      if theSelection is {} then
+        return
+      else if (theSelection count) is equal to 1 then
+        if (kind of the first item of theSelection) contains "image" then
+          return the POSIX path of (theSelection as alias)
+        end if
+      else
+        set thePaths to {}
+        repeat with i from 1 to (theSelection count)
+          if (kind of (item i of theSelection)) contains "image" then
+            copy (POSIX path of (item i of theSelection as alias)) to end of thePaths
+          end if
+        end repeat
+        return thePaths
+      end if
+    end tell`
+  );
+
+  return selectedImages.split(", ");
+};
```

Comment:
You can get this from Raycast with [getSelectedFinderItems](https://developers.raycast.com/api-reference/environment#getselectedfinderitems)

Reply:
I'd like to use the items' data kind, which to my understanding is not necessarily the same as the file extension, to filter the list of selected items. I don't think that is possible with Raycast's `getSelectedFinderItems` -- correct me if I'm wrong though.

Reply:
No that's correct, you can use this if you want 😊

let me know 😊